src/binwalk/magic/firmware

```diff
@@ -468,11 +468,13 @@
 # Found in DIR-100 firmware
 0       string      AIH0        AIH0 firmware header, header size: 48,
 >12     ubelong     0           {invalid}
->12     ubelong+48  x           {size:%d}
+>12     ubelong     >0x4000000  {invalid}
+>12     ubelong     x           {size:%d}
```

Comment:
I don't think so, unless maybe they changed the format between different firmware versions? Here's an example:

```
$ binwalk DIR100_FW401WWB02.bin 

DECIMAL       HEXADECIMAL     DESCRIPTION
--------------------------------------------------------------------------------
0             0x0             AIH0 firmware header, header size: 48, data size: 275464, executable code, load address: 0x80000100, version: "4.01"
48            0x30            LZMA compressed data, properties: 0x5D, dictionary size: 33554432 bytes, uncompressed size: 996072 bytes
275512        0x43438         AIH0 firmware header, header size: 48, data size: 172524, version: "4.01"
...
```

If you take the reported size in the first AIH0 header (275464) and add the header size of 48 to it, you get 275512, which is exactly the start of the second AIH0 header in this firmware.